test/rules/require-capitalized-constructors.js

```diff
@@ -4,25 +4,59 @@ var assert = require('assert');
 describe('rules/require-capitalized-constructors', function() {
     var checker;
 
-    beforeEach(function() {
-        checker = new Checker();
-        checker.registerDefaultRules();
-        checker.configure({ requireCapitalizedConstructors: true });
-    });
+    describe('with `true` value', function() {
+        beforeEach(function() {
+            checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({ requireCapitalizedConstructors: true });
+        });
 
-    it('should report uncapitalized construction', function() {
-        assert(checker.checkString('var x = new y();').getErrorCount() === 1);
-    });
+        it('should report uncapitalized construction', function() {
+            assert(checker.checkString('var x = new y();').getErrorCount() === 1);
+        });
 
-    it('should not report capitalized construction', function() {
-        assert(checker.checkString('var x = new Y();').isEmpty());
-    });
+        it('should not report capitalized construction', function() {
+            assert(checker.checkString('var x = new Y();').isEmpty());
+        });
 
-    it('should not report member expression construction', function() {
-        assert(checker.checkString('var x = new ns.y();').isEmpty());
+        it('should not report member expression construction', function() {
+            assert(checker.checkString('var x = new ns.y();').isEmpty());
+        });
+
+        it('should not report construction with "this" keyword', function() {
+            assert(checker.checkString('var x = new this();').isEmpty());
+        });
     });
 
-    it('should not report construction with "this" keyword', function() {
-        assert(checker.checkString('var x = new this();').isEmpty());
+    describe('with `Array` value', function() {
+        beforeEach(function() {
+            checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({
+                requireCapitalizedConstructors: {
+                    allExcept: ['require']
+                }
+            });
+        });
+
+        it('should report uncapitalized construction', function() {
```

Comment:
Maybe extract the cases that are redundant between the describes into a `checkBaseCases()` function? That'll make it clearer which cases are unique to the different configuration value.